5.5/docker-entrypoint.sh

```diff
@@ -33,23 +58,35 @@ if [ "$1" = 'mysqld' ]; then
 		EOSQL
 		
 		if [ "$MYSQL_DATABASE" ]; then
-			echo "CREATE DATABASE IF NOT EXISTS \`$MYSQL_DATABASE\` ;" >> "$tempSqlFile"
+			echo "CREATE DATABASE IF NOT EXISTS \`"$MYSQL_DATABASE"\` ;" >> "$tempSqlFile"
 		fi
 		
 		if [ "$MYSQL_USER" -a "$MYSQL_PASSWORD" ]; then
-			echo "CREATE USER '$MYSQL_USER'@'%' IDENTIFIED BY '$MYSQL_PASSWORD' ;" >> "$tempSqlFile"
+			echo "CREATE USER '"$MYSQL_USER"'@'%' IDENTIFIED BY '"$MYSQL_PASSWORD"' ;" >> "$tempSqlFile"
 			
 			if [ "$MYSQL_DATABASE" ]; then
-				echo "GRANT ALL ON \`$MYSQL_DATABASE\`.* TO '$MYSQL_USER'@'%' ;" >> "$tempSqlFile"
+				echo "GRANT ALL ON \`"$MYSQL_DATABASE"\`.* TO '"$MYSQL_USER"'@'%' ;" >> "$tempSqlFile"
 			fi
 		fi
 		
 		echo 'FLUSH PRIVILEGES ;' >> "$tempSqlFile"
 		
-		set -- "$@" --init-file="$tempSqlFile"
+		chown -R mysql:mysql "$DATADIR"
+		mysql -uroot < $tempSqlFile
+
+		rm -f $tempSqlFile
+		kill $(cat $PIDFILE)
+                for i in $(seq 30 -1 0); do
+		    [ -S $SOCKET ] || break
+		    echo 'MySQL init process in progress...'
+		    sleep 1
+		done
+                if [ $i = 0 ]; then
+                    echo >&2 'MySQL hangs during init process.'
+                    exit 1
+                fi
+		echo 'MySQL init process done. Ready for start up.'
 	fi
-	
-	chown -R mysql:mysql "$DATADIR"
```

Comment:
Permissions are now only set during initialization. Would this be a problem when using an already initialized data-dir (e.g. bind-mounted)?